### PR TITLE
Generate riff-distro documentation

### LINKED ISSUE #881: Check in riff-distro docs
This should be done analogously to the `riff` docs.

The docs could appear either in `docs/distro` or `distro/docs`. The former may make them easier to find; the latter could provide better separation of docs for different audiences.

### PR DESCRIPTION
**Pull request by fbiville**
Fixes #881

### CODE REVIEW

**Comment by fbiville** [comment]
For early review, I still plan to reconcile the two slightly different implementations.

File: distro/commands/docs.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		Use:    "docs",
+		Short:  "generate riff-distro command documentation",
+		Hidden: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			fileinfo, err := fs.Stat(directory)
+			if os.IsNotExist(err) {
+				if err = fs.Mkdir(directory, 0744); err != nil {
+					return err
+				}
+				fileinfo, err = fs.Stat(directory)
```
Comment (fbiville) [inline distro/commands/docs.go]: Without the line above, the doc generation will fail if the docs folder is not present, which makes sense.
I do not understand how the other impl does not need that.

File: distro/commands/docs.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			}
+
+			if err != nil {
```
Comment (fbiville) [inline distro/commands/docs.go]: this bit is missing in the pre-existing implementation, thus discarding possible os.Stat errors

File: distro/commands/wiring_test.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		It("including `riff-distro` docs", func() {
+			errMsg := "`%s` should be wired to root command"
+			Expect(find(rootCommand, "docs")).NotTo(BeNil(), fmt.Sprintf(errMsg, "docs"))
+		})
+
+	})
+
+})
+
+func find(command *cobra.Command, names ...string) *cobra.Command {
```
Comment (fbiville) [inline distro/commands/wiring_test.go]: it's a copy of the other wiring test, I will define it in a single place instead

File: cmd/commands/docs.go [hunk 4]
```diff
@@ -34,19 +34,23 @@ func Docs(rootCmd *cobra.Command) *cobra.Command {
...
 		},
 	}
 	docsCmd.Flags().StringVarP(&directory, "dir", "d", "docs", "the output directory for the docs.")
 	return docsCmd
 }
+
+func GenerateDocs(rootCommand *cobra.Command, directory string, fs Filesystem) error {
+	fileinfo, err := fs.Stat(directory)
+	if os.IsNotExist(err) {
+		if err = fs.Mkdir(directory, 0744); err != nil {
```
Comment (ericbottard) [inline cmd/commands/docs.go]: Might as well call this (and use in the underlying implementation) `MkDirAll`

File: cmd/commands/wiring.go [hunk 5]
```diff
@@ -220,5 +220,12 @@ func installKubeConfigSupport(command *cobra.Command, client *core.Client, kc *c
 		}
 		return nil
 	}
+}
 
+func FindSubcommand(command *cobra.Command, names ...string) *cobra.Command {
```
Comment (ericbottard) [inline cmd/commands/wiring.go]: Can this be made private ?

Comment (fbiville) [reply cmd/commands/wiring.go]: It's used in `github.com/projectriff/riff/cmd/commands` and `github.com/projectriff/riff/distro/commands`

Comment (fbiville) [reply cmd/commands/wiring.go]: It's used in `github.com/projectriff/riff/cmd/commands` and `github.com/projectriff/riff/distro/commands`

Comment (fbiville) [reply cmd/commands/wiring.go]: It's used in `github.com/projectriff/riff/cmd/commands` and `github.com/projectriff/riff/distro/commands`

Comment (fbiville) [reply cmd/commands/wiring.go]: It's used in `github.com/projectriff/riff/cmd/commands` and `github.com/projectriff/riff/distro/commands`

File: cmd/commands/docs.go (hunk 4 above)
Comment (fbiville) [reply cmd/commands/docs.go]: The UI does not show anything but it's now done (cf. status.github.com, they currently have DB issues)

File: cmd/commands/wiring.go (hunk 5 above)
Comment (ericbottard) [reply cmd/commands/wiring.go]: Maybe move to `cobra.go` then

Comment (ericbottard) [reply cmd/commands/wiring.go]: Maybe move to `cobra.go` then

Comment (fbiville) [reply cmd/commands/wiring.go]: 👍

**Comment by ericbottard** [comment]
Merged as 73bd71f4

**Pull request closed without merging**